src/main/java/com/googleinterns/zoomtube/data/Comment.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.googleinterns.zoomtube.data;
+
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.repackaged.com.google.common.base.Optional;
+import com.google.auto.value.AutoValue;
+import java.util.Date;
+import javax.annotation.Nullable;
+
+/** Contains data related to a comment in a discussion. */
+@AutoValue
+public abstract class Comment {
+  public static final String ENTITY_KIND = "Comment";
+  public static final String PROP_LECTURE = "lecture";
+  public static final String PROP_PARENT = "parent";
+  public static final String PROP_TIMESTAMP = "timestamp";
+  public static final String PROP_AUTHOR = "author";
+  public static final String PROP_CONTENT = "content";
+  public static final String PROP_CREATED = "created";
+
+  public static Comment create(Key key, Key lecture, Optional<Key> parent, double timestamp,
+      String author, String content, Date created) {
+    return new AutoValue_Comment(key, lecture, parent, timestamp, author, content, created);
+  }
+
+  /**
+   * Creates a {@code Comment} from a datastore {@link com.google.appengine.api.datastore.Entity}
+   * using the property names defined in this class.
+   */
+  public static Comment fromEntity(Entity entity) {
+    Key key = entity.getKey();
+    Key lecture = (Key) entity.getProperty(PROP_LECTURE);
+    Optional<Key> parent = Optional.fromNullable((Key) entity.getProperty(PROP_PARENT));
+    double timestamp = (double) entity.getProperty(PROP_TIMESTAMP);
+    String author = (String) entity.getProperty(PROP_AUTHOR);
+    String content = (String) entity.getProperty(PROP_CONTENT);
+    Date created = (Date) entity.getProperty(PROP_CREATED);
+    return Comment.create(key, lecture, parent, timestamp, author, content, created);
+  }
+
+  /** Returns the comment's Datastore entity key. */
+  public abstract Key key();
+
+  /** Every comment is about a lecture, this returns the lecture's Datastore entity key. */
+  public abstract Key lecture();
+
+  /**
+   * Returns the key of the comment this is a reply to, or {@code Optional.empty()} if this
+   * isn't a reply.
+   */
+  public abstract Optional<Key> parent();
+
+  /** Returns the timestamp in the video this comment is referencing. */
+  public abstract double timestamp();
```

Comment:
Please read https://docs.google.com/document/d/1cGX5gkcNdnwGCU3ToxKiuhpyVX17VkG-VzX01UJ_2RY/edit and find the best way of representing the timestamp. I do not want this to be a source for bugs.

Reply:
See discussion in chat

Reply:
Changed to date